src/record.cc

```diff
@@ -107,6 +220,19 @@ namespace warc2text {
     }
 
     int Record::cleanPayload(const util::umap_tag_filters& tagFilters){
+        if(WARCcontentType.find("application/http") == std::string::npos && !bdf_zip && textContentTypes.find(getHTTPcontentType()) == textContentTypes.end())
+            return util::NOT_VALID_RECORD;
+
+        std::string content_type;
+        if (HTTPheader.count("content-type") == 1)
+            content_type = HTTPheader["content-type"];
+        else
+            content_type = "";
```

Comment:
this is already stored in `this->cleanHTTPcontentType`

Reply:
Right! And if there is no content-type in the WARC Record header or the HTTP header, that cleanHTTPcontentType is already empty. Thanks!